File: Documentation/API/Parsers/Chart/ReadNotesFromChartData.js
import { readFileSync } from 'node:fs';

import RhythmGameUtilities from '@rhythm-game-utilities/core';

const contents = readFileSync('./tests/Mocks/song.chart', 'utf-8');

const notes = RhythmGameUtilities.ReadNotesFromChartData(contents, 'Expert');

for (let note of notes) {
  console.log(`${note.position} ${note.handPosition}`);
}

File: Documentation/API/Parsers/Chart/ReadNotesFromChartData.lua
---@type RhythmGameUtilities
local rhythmgameutilities = require("rhythmgameutilities")

local file = io.open("./tests/Mocks/song.chart", "r")

if file then
    local content = file:read("*a")

    local notes = rhythmgameutilities.read_notes_from_chart_data(content, Difficulty.Expert);

    for _, note in pairs(notes) do
        print(note["position"] .. " " .. note["hand_position"])
    end
end
